controllers/pwa.js

```diff
@@ -163,30 +160,141 @@ router.post('/add', (req, res, next) => {
  * GET /pwas/:id
  *
  * Display a PWA.
+ * Stores the latest detail page in memcached.
  */
 router.get('/:pwa', (req, res, next) => {
+  const pwaId = req.params.pwa;
   const contentOnly = false || req.query.contentOnly;
-  pwaLib.find(req.params.pwa)
+  const url = req.originalUrl;
+  cache.get(url)
+    .then(cachedPage => {
+      console.log('Served from memcache: ' + url);
+      res.send(cachedPage);
+      return cachedPage;
+    })
+    .catch(_ => {
+      console.log('Not in memcache: ' + url);
+    })
+    .then(cachedPage => {
```

Comment:
Looks like this is always rendering the page again, even if it's cached. We're also removing from the cache when the PWA is updated. Maybe this is not necessary.